stdlib/public/TensorFlow/Dataset.swift

```diff
@@ -0,0 +1,141 @@
+//===-- Dataset.swift -----------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// The dataset API.
+//
+//===----------------------------------------------------------------------===//
+
+/// Represents a potentially large set of elements.
+///
+/// A `SingleValueDataset` can be used to represent an input pipeline as a
+/// collection of element tensors.
+@_fixed_layout
+public struct SingleValueDataset {
+  // NOTE: SingleValueDataset is currently specialzed on `Float` to work around
+  // deabstraction bug SR-8463. This is to avoid indirect passing in SIL. When
```

Comment:
Is this the correct bug number?

Reply:
It is. When SingleValueDataset or SingleValueDatasetIterator is on the stack, const eval won’t work.